ui/selectors/multichain-accounts/multichain-accounts-selectors.test.ts

```diff
@@ -0,0 +1,250 @@
+import mockState from '../../../test/data/mock-state.json';
+import {
+  getAccountTree,
+  getWalletsWithAccounts,
+} from './multichain-accounts-selectors';
+import { MultichainAccountsState } from './multichain-accounts-selectors.types';
+
+describe('Multichain Accounts Selectors', () => {
+  describe('getAccountTree', () => {
+    it('returns the account tree', () => {
+      const result = getAccountTree(
+        mockState as unknown as MultichainAccountsState,
+      );
+
+      expect(result).toStrictEqual(mockState.metamask.accountTree);
+    });
+  });
+
+  describe('getWalletsWithAccounts', () => {
+    it('returns wallets with accounts and their metadata', () => {
+      const result = getWalletsWithAccounts(mockState);
+
+      expect(result).toStrictEqual({
```

Comment:
For other reviewers: I had another suggestion to avoid duplicating a big chunk of the state, but it seems this new refactor also impacts some other pending PRs (related to those new selectors). For now, we'll go with this.

Diff is here:
- https://github.com/MetaMask/metamask-extension/compare/dd/add-multichain-account-selectors...cc/add-multichain-account-selectors